checkov/cdk/checks/python/TransferServerIsPublic.yaml

```
@@ -0,0 +1,14 @@
+metadata:
+  version: 0.2
+  approach: define failing
+  id: CKV_AWS_164
+  name: Ensure Transfer Server is not exposed publicly.
+  category: GENERAL_SECURITY
+scope:
+  languages:
+    - python
+definition:
+  pattern: aws_cdk.aws_transfer.CfnServer(<ANY>)
+  conditions:
+    - not_pattern: aws_cdk.aws_transfer.CfnServer(<ANY>, endpoint_type='VPC', <ANY>)
+    - not_pattern: aws_cdk.aws_transfer.CfnServer(<ANY>, endpoint_type='VPC_ENDPOINT' , <ANY>)
```

I guess it can be something like:
```suggestion
    - not_pattern: aws_cdk.aws_transfer.CfnServer(<ANY>, endpoint_type=$VPC_STRING, <ANY>)
      conditions: 
        or: 
          - metavariable: $VPC_STRING
             comparison: "VPC  
          - metavariable: $VPC_STRING
             comparison: "VPC_ENDPOINT"  
      
```

---

we don't support this syntax, i opened a ticket about that last week (to add support for conditions inside not_pattern)